tests/tests_pytorch/test_cli.py

```diff
@@ -1860,6 +1860,34 @@ def test_lightning_cli_args_and_sys_argv_warning():
         LightningCLI(TestModel, run=False, args=["--model.foo=789"])
 
 
+def test_add_class_args_required_false_skips_addition(tmp_path):
```

Comment:
@KAVYANSHTYAGI you did not run the test to check whether the code was working. I know this because it is not possible that it works. When you open a pull request it would be good that beforehand you run the tests. And the test should fail without the code changes and succeed with them.